### PR TITLE
Replace outdated actions in release automation workflow

### LINKED ISSUE #5111: Replace semver-action in release automation workflow
### Problem

The `semver-action` causes a lot of warnings

```
Warning: The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
```

We need to find an alternative before `set-output` is disabled

### Solution

- Find an alternative github action with functionality parity

### Alternatives

fork the github `semver-action` action and make the recommended changes

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

This PR replaces outdated actions in release automation workflow

- Replace semver action
- Replace milestone action
- Switch to official release note generator action

**Related issue(s)**:

Fixes #5111 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: .github/workflows/release-automation.yml [hunk 1]
```diff
@@ -22,30 +22,26 @@ jobs:
...
+        uses: madhead/semver-utils@latest
         with:
-          bump: preminor
+          lenient: false
           version: ${{ github.event.inputs.version }}
 
-      - name: Retrieve the Milestone
-        id: milestone
-        if: ${{ steps.version_parser.outputs.pre-release == '' }}
-        uses: julbme/gh-action-manage-milestone@v1
```
Comment (xin-hedera) [inline .github/workflows/release-automation.yml:33]: the milestone action also uses the to-be-deprecated `set-output` command

File: .github/workflows/release-automation.yml [hunk 2]
```diff
@@ -87,15 +83,24 @@ jobs:
       - name: Gradle Release
         uses: gradle/gradle-build-action@v2
         with:
-          arguments: release -Pversion=${{ steps.version_parser.outputs.version }}
+          arguments: release -Pversion=${{ env.VERSION }}
+
+      - name: Close the Milestone
+        if: ${{ steps.version_parser.outputs.prerelease == '' }}
+        id: milestone
+        uses: Akkjon/close-milestone@v2.1.0
```
Comment (xin-hedera) [inline .github/workflows/release-automation.yml:91]: the `close-milestone` action doesn't have a get-milestone-id-only function, so have to close it and get the milestone id in its output

File: .github/workflows/release-automation.yml [hunk 3]
```diff
@@ -22,30 +22,26 @@ jobs:
...
-        with:
-          title: ${{ steps.version_parser.outputs.version }}
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
       - name: Set Release Environment Variables
         run: |
-          NEXT_VERSION_SNAPSHOT=${{ format('{0}.{1}.0-SNAPSHOT', steps.version_parser.outputs.major, steps.version_parser.outputs.next-minor) }}
+          PREMINOR_VERSION=${{ steps.version_parser.outputs.inc-preminor }}
+          NEXT_VERSION_SNAPSHOT=${PREMINOR_VERSION//-0/-SNAPSHOT}
```
Comment (xin-hedera) [inline .github/workflows/release-automation.yml:33]: `semver-utils`'s `inc-premiro` output is in the format `0.80.0-0` so we need to replace the `-0` suffix

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by mgoelswirlds — APPROVED** [review]

**Pull request merged**